src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+import { $, ply, TimeRange, Expression, Dataset } from 'plywood';
+import {
+  MeasureModeNeeded,
+  Manifest,
+  Measure,
+  VisualizationProps,
+  DatasetLoad,
+  Splits,
+  DataSource,
+  Colors,
+  Resolve,
+  Essence
+} from '../../../common/models/index';
+
+import { SPLIT, SEGMENT, TIME_SEGMENT } from '../../config/constants';
+
+
+export interface BaseVisualizationState {
+  datasetLoad?: DatasetLoad;
+  dragStartTime?: Date;
+  dragTimeRange?: TimeRange;
+  roundDragTimeRange?: TimeRange;
+  dragOnMeasure?: Measure;
+  scrollLeft?: number;
+  scrollTop?: number;
+  hoverTimeRange?: TimeRange;
```

Comment:
`hoverTimeRange: TimeRange` definitely should not be in here. That is purely a time series thing. Maybe we should rename it to `hoverValue: PlywoodValue` or take it out of the base class?